src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs

```diff
@@ -344,22 +348,24 @@ public void Login()
         public bool OpenRepository()
         {
             var old = Repositories.FirstOrDefault(x => x.Equals(Holder.ActiveRepo));
-            // open the solution selection dialog when the user wants to switch to a different repo
-            // since there's no other way of changing the source control context in VS
             if (!Equals(SelectedRepository, old))
             {
-                if (ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1)))
+                var opened = vsServices.TryOpenRepository(SelectedRepository.LocalPath);
+                if (!opened)
                 {
-                    ServiceProvider.TryGetService<ITeamExplorer>()?.NavigateToPage(new Guid(TeamExplorerPageIds.Home), null);
-                    return true;
+                    // Fall back to making the user select a solution to open.
+                    opened = ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1));
                 }
-                else
+
+                if (!opened)
                 {
-                    SelectedRepository = old;
                     return false;
```

Comment:
I added some logging for `SelectedRepository` and it seems to change when the user single-clicks on the list (or scrolls up and down using the up/down arrows). I'm not sure it's terribly meaningful beyond being a cursor for the user. Do you really think it needs to  be reset?
![image](https://cloud.githubusercontent.com/assets/11719160/22839938/69660d1e-efc3-11e6-872e-c62f697ea786.png)

Please give me a clue if I'm missing something! 😉